### ISSUE TITLE
 keyboard focus is not clearly visible on selected checkbox controls, while navigating using tab key.

### ISSUE THREAD

**Issue by v-kumarsatya** [IgniteUI/igniteui-angular#13519]
If the focus indicator is not clearly visible, users may struggle to interact with checkboxes effectively using only the keyboard. This can result in users being unable to accurately select or deselect checkboxes, impacting their ability to interact with forms or perform important actions.

Actual Result:
keyboard focus is not clearly visible on selected checkbox controls, while navigating using tab key.
Observation: Rectangular keyboard focus is not visible on selected 'Select all, NA' etc. checkbox controls but it is visible on not selected checkbox controls.
Expected Result:
keyboard focus should be clearly visible on selected checkbox controls, while navigating using tab key.

source link- [Excel Style Filtering in Angular Data Grid- Ignite UI for Angular (infragistics.com)](https://www.infragistics.com/products/ignite-ui-angular/angular/components/grid/excel-style-filtering)

**Comment by mddragnev** [comment]
@ChronosSF, @dkamburov  Couple of things worth mentioning here which we should firstly decide before starting to implement this:
1. It was set tabindex=-1 (resulting in no navigation) on the checkboxes on purpose but there is no information in the spec about it.
2. If we want to make it navigatable we should firstly decide what keys we should support and what should be focusable
3. We can see how the combo (looks almost similar) works with the igxForOf directive:
 with Tab we focus the first item and then only arrowUp and arrowDown works (home,end too) (which sounds good enough for me)
4. Because the list items in the ESF are using the virtualization we should implement a custom handling so this is probably a future request and not a bug. But 100% before that we should update the spec when we decide exactly how it would work.

So can we guys decide these questions and updating the spec for the ESF before implementing it?

**Comment by ChronosSF** [comment]
Implementation wise it's a FR because it was never considered but UX wise it's a critical bug and should've never been released without any accessibility. Please fix it as a bug.

As for the implementation - research ARIA specs and best practices for lists and implement it in that way. Good start is https://www.w3.org/WAI/ARIA/apg/patterns/listbox/ but it might be an overkill.

You could also check the combobox dropdown as it must have ARIA considerations and it's also implementation that is incredibly similar to that ESF value list. If you could exctract an internal component to be used for both cases, that would be ideal.
